### PR TITLE
Add new project: kcodecs

### PR DESCRIPTION
**Pull request by tsdgeos**
I'm part of KDE's security team 

https://www.kde.org/info/security/

### CODE REVIEW

**Comment by inferno-chromium** [comment]
Putting this in sheriff's queue.

**Comment by Dor1s** [comment]
@tsdgeos, did you follow the New Project Guide, especially the [Testing Locally](https://github.com/google/oss-fuzz/blob/master/docs/new_project_guide.md#testing-locally) section?

**Comment by tsdgeos** [comment]
> @tsdgeos, did you follow the New Project Guide, especially the [Testing Locally](https://github.com/google/oss-fuzz/blob/master/docs/new_project_guide.md#testing-locally) section?

Yes, this is the 3rd project i submit, i have tested them all.

**Comment by Dor1s** [comment]
> Yes, this is the 3rd project i submit/contribute to, i have tested them all.

Fantastic! Just want to clarify that as quite often contributors don't really test stuff properly and things break later on :)

File: projects/kcodecs/build.sh [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+# Disable compressing rcc files, triggers a warning in the memory sanitizer that i'm not sure is valid. TODO investigate properly
+sed -i -e "s/DEFINES += QT_RCC QT_NO_CAST_FROM_ASCII QT_NO_FOREACH/DEFINES += QT_NO_COMPRESS QT_RCC QT_NO_CAST_FROM_ASCII QT_NO_FOREACH/g" src/tools/rcc/rcc.pro
+./configure --glib=no --libpng=qt -opensource -confirm-license -static -no-opengl -no-icu -platform linux-clang-libc++ -v
+cd src
+../bin/qmake -o Makefile src.pro
+make sub-corelib -j$(nproc)
+
+cd $SRC
+cd kcodecs
+cmake . -DBUILD_SHARED_LIBS=OFF -DCMAKE_BUILD_TYPE=Debug -DCMAKE_INSTALL_PREFIX=$SRC/qtbase
```
[the remark this answers is not in the document]
Comment (tsdgeos) [reply projects/kcodecs/build.sh]: Done it's not triggering asserts.

File: projects/kcodecs/build.sh [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+make -j$(nproc) VERBOSE=1
+
+
+$CXX $CXXFLAGS -fPIC -std=c++11 $SRC/kcodecs_fuzzer.cc -o $OUT/kcodecs_fuzzer -I $SRC/qtbase/include/QtCore/ -I $SRC/qtbase/include/  -I $SRC/kcodecs/src -I $SRC/kcodecs/src/probers -L $SRC/qtbase/lib -L $SRC/kcodecs/lib -lQt5Core -lm -lqtpcre2 -ldl -lpthread -lFuzzingEngine -lKF5Codecs
```
[the remark this answers is not in the document]
Comment (tsdgeos) [reply projects/kcodecs/build.sh]: it seems like you need a nicer editor that helps with long lines readability, but i'll try to do it even if that means have to sit during a 20 minute build wasting my time to make sure i didn't break the build while making a line arguably nicer

File: projects/kcodecs/kcodecs_fuzzer.cc [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+        new kencodingprober::nsUniversalDetector(),
+        new kencodingprober::ChineseGroupProber(),
+        new kencodingprober::nsEscCharSetProber(),
+        new kencodingprober::nsEUCKRProber(),
+        new kencodingprober::nsMBCSGroupProber(),
+        new kencodingprober::nsSJISProber(),
+        new kencodingprober::UnicodeGroupProber()
+    };
+
+    for (kencodingprober::nsCharSetProber *p : probers) {
```
[the remark this answers is not in the document]
Comment (tsdgeos) [reply projects/kcodecs/kcodecs_fuzzer.cc]: I would rather not do that.

**Comment by Dor1s** [comment]
Thanks, @tsdgeos! SInce it's not the first project you're adding, you might've already heard of [Ideal Integration](https://github.com/google/oss-fuzz/blob/master/docs/ideal_integration.md). An extra benefit of it is that maintainers store the sources / build rules in their own repo and don't need to comply with our code review requests :) That also allows the maintainers to apply for a greater [integration reward](https://opensource.googleblog.com/2017/05/oss-fuzz-five-months-later-and.html) and saves some time on our end.

**Comment by tsdgeos** [comment]
I'm not sure the test of developers would want to have the sources in the upstream repo, i'll have to talk with them.

Thanks for mentioning it :)

**Pull request merged**